Sources/Sentry/SentryOptions.m

```diff
@@ -360,7 +360,7 @@ - (BOOL)validateOptions:(NSDictionary<NSString *, id> *)options
         self.maxCacheItems = [options[@"maxCacheItems"] unsignedIntValue];
     }
 
-    if ([options[@"cacheDirectoryPath"] isKindOfClass:[NSString class]]) {
+    if ([SentryOptionsValidator isCacheDirectoryPathValidWithPath:options[@"cacheDirectoryPath"]]) {
```

Comment:
`h`: What drove you to duplicate this logic all over the SDK? Wouldn't one check in a custom setter of `cacheDirectoryPath` be sufficient?

Reply:
This was the only validation logic I could find regarding options validation, and it only validates casting of the type-erased value from the Hybrid SDKs.

As this was my starting point to implement a validator class, I moved the logic away from here.

I'm open for discussion if this should not be changed.

Regarding the duplicating all over the SDK, there is a TODO in this PR to decide where the appropriate place to add validation is, therefore I'll add a comment directly to the PR instead.